scripts/pylib/twister/twisterlib/handlers.py

```diff
@@ -395,7 +395,7 @@ def monitor_serial(self, ser, halt_event, harness):
                 sl = serial_line.decode('utf-8', 'ignore').lstrip()
                 logger.debug("DEVICE: {0}".format(sl.rstrip()))
 
-                log_out_fp.write(sl)
+                log_out_fp.write(strip_ansi_sequences(sl))
```

Comment:
> ANSI color is great for viewing output in a terminal, but currently it also gets written out to the handler.log file. Text editors usually don't render these, 

well, not only ANSI escapes, but some non-printable bytes appear at serial lines as well making text editors crazy.

The proposed change filters ANSI escapes for the log file only whereas main problems are actually at the harness pattern recognition.

I also think this filtering have to be configurable, no filtering by default.
For example as a hardware-map bool attribute applied to `serial` and `serial_pty`.